## Introduction

In this project, working with data from the CIA World Factbook, a compendium of statistics about all of the countries on Earth. The Factbook contains demographic information.

In [3]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

Introduced the name of the table and what the table looked like in the "factbook.db" database. Run the queries below to get this information directly

In [8]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type = 'table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

For the next run the query to return information on the tables in the database and show first five rows to explore database.

## Overview of the Data

In [10]:
%%sql
SELECT *
  FROM facts
 LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Here are the descriptions for some of the columns:

 - name --the name of the country.
 - area --the country's total area (both land and water).
 - area_land --the country's land area in square kilometers.
 - area_water --the country's waterarea in square kilometers.
 - population --the country's population.
 - population_growth --the country's population growth as a percentage.
 - birth_rate --the country's birth rate, or the number of births per year per 1,000 people.
 - death_rate --the country's death rate, or the number of death per year per 1,000 people.

## Summary Statistics

In [13]:
%%sql
SELECT MIN(population) AS min_pop,
       MAX(population) AS max_pop,
       MIN(population_growth) AS min_pop_growth,
       MAX(population_growth) AS max_pop_growth
  FROM facts;

Done.


[(0, 7256490011, 0.0, 4.02)]

After summarize statistics calculation with facts table

 - There's some country with population of 0.
 - There's some country with population 7256490011 (7.2 Billion people)
 - There's some country with population growth of 0.
 - There's some country with population growth of 4.02

## Exploring Outliers

In [15]:
%%sql
SELECT name, population, birth_rate, death_rate
  FROM facts
 WHERE population = (SELECT MIN(population)
                       FROM facts);

Done.


[('Antarctica', 0, None, None)]

Try to figure out what the country that have 0 population and finally found that country is 'Antarctica'!!

In [16]:
%%sql
SELECT name, population, birth_rate, death_rate
  FROM facts
 WHERE population = (SELECT MAX(population)
                       FROM facts);

Done.


[('World', 7256490011, 18.6, 7.8)]

And try to figure out what the country that have 7.2+ Billion people and found that country is the total result from the whole world!!

Now that we know this, we should recalculate the summary statistics we calculated earlier, while excluding the row for the whole world.

## Summary Statistics Revisited

In [19]:
%%sql
SELECT MIN(population),
       MAX(population),
       MIN(population_growth),
       MAX(population_growth)
  FROM facts
 WHERE name <> 'World';

Done.


[(0, 1367485388, 0.0, 4.02)]

There's a country whose population closes in on 1.4 billion!

In [27]:
%%sql
SELECT name
  FROM facts
 WHERE population = (SELECT MAX(population)
                       FROM facts
                      WHERE name <> 'World');
                

Done.


[('China',)]

Ofcourse, China is the answer!

## Exploring Average Population and Area

Let's explore density. Density depends on the population and the country's area. Let's look at the average values for these two columns.

In [32]:
%%sql
-- Find out the average of population and the country's area
SELECT ROUND(AVG(population), 2) AS avg_pop,
       ROUND(AVG(area), 2) AS avg_area
  FROM facts
 WHERE name <> 'World';

Done.


[(32242666.57, 555093.55)]

In [100]:
%%sql
/* 1) Find out the country that have the population is above average.
   2) Find out the country that have the area is below average.
*/

SELECT *
  FROM facts
 WHERE population > (SELECT AVG(population)
                       FROM facts
                      WHERE name <> 'World'
                    )
       
   AND area < (SELECT AVG(area)
                 FROM facts
                WHERE name <> 'World'
              );
 LIMIT 10

Done.
(sqlite3.OperationalError) near "LIMIT": syntax error
[SQL: LIMIT 10]
(Background on this error at: http://sqlalche.me/e/13/e3q8)


These country in the table above is the country that have density from both the number of population and number of country's area.

But, Actually need to recalculate with the "area_land" column with the other way and the people live in the land area not water area.

In [62]:
%%sql
SELECT name, population, area_land,
       population / area_land AS pop_per_area
  FROM facts
 WHERE name <> 'World'
 ORDER BY pop_per_area DESC;

Done.


[('Macau', 592731, 28, 21168),
 ('Monaco', 30535, 2, 15267),
 ('Singapore', 5674472, 687, 8259),
 ('Hong Kong', 7141106, 1073, 6655),
 ('Gaza Strip', 1869055, 360, 5191),
 ('Gibraltar', 29258, 6, 4876),
 ('Bahrain', 1346613, 760, 1771),
 ('Maldives', 393253, 298, 1319),
 ('Malta', 413965, 316, 1310),
 ('Bermuda', 70196, 54, 1299),
 ('Bangladesh', 168957745, 130170, 1297),
 ('Sint Maarten', 39689, 34, 1167),
 ('Guernsey', 66080, 78, 847),
 ('Jersey', 97294, 116, 838),
 ('Taiwan', 23415126, 32260, 725),
 ('Barbados', 290604, 430, 675),
 ('Mauritius', 1339827, 2030, 660),
 ('Aruba', 112162, 180, 623),
 ('Lebanon', 6184701, 10230, 604),
 ('Saint Martin', 31754, 54, 588),
 ('San Marino', 33020, 61, 541),
 ('Rwanda', 12661733, 24668, 513),
 ('Korea, South', 49115196, 96920, 506),
 ('Netherlands', 16947904, 33893, 500),
 ('West Bank', 2785366, 5640, 493),
 ('Nauru', 9540, 21, 454),
 ('India', 1251695584, 2973193, 420),
 ('Burundi', 10742276, 25680, 418),
 ('Tuvalu', 10869, 26, 418),
 ('Puerto Rico', 3598357, 8870, 405),
 ('Marshall Islands', 72191, 181, 398),
 ('Israel', 8049314, 20330, 395),
 ('Belgium', 11323973, 30278, 374),
 ('Haiti', 10110019, 27560, 366),
 ('Comoros', 780971, 2235, 349),
 ('Japan', 126919659, 364485, 348),
 ('Sri Lanka', 22053488, 64630, 341),
 ('Philippines', 100998376, 298170, 338),
 ('Curacao', 148406, 444, 334),
 ('Grenada', 110694, 344, 321),
 ('Vietnam', 94348835, 310070, 304),
 ('Virgin Islands', 103574, 346, 299),
 ('Guam', 161785, 544, 297),
 ('El Salvador', 6141350, 20721, 296),
 ('American Samoa', 54343, 199, 273),
 ('Jamaica', 2950210, 10831, 272),
 ('Saint Lucia', 163922, 606, 270),
 ('United Kingdom', 64088222, 241930, 264),
 ('Saint Vincent and the Grenadines', 102627, 389, 263),
 ('Pakistan', 199085847, 770875, 258),
 ('Trinidad and Tobago', 1222363, 5128, 238),
 ('Liechtenstein', 37624, 160, 235),
 ('Germany', 80854408, 348672, 231),
 ('British Virgin Islands', 33454, 151, 221),
 ('Luxembourg', 570252, 2586, 220),
 ('Nepal', 31551305, 143351, 220),
 ('Dominican Republic', 10478756, 48320, 216),
 ('Cayman Islands', 56092, 264, 212),
 ('Italy', 61855120, 294140, 210),
 ('Antigua and Barbuda', 92436, 442, 209),
 ('Korea, North', 24983205, 120408, 207),
 ('Seychelles', 92430, 455, 203),
 ('Switzerland', 8121830, 39997, 203),
 ('Sao Tome and Principe', 194006, 964, 201),
 ('Nigeria', 181562056, 910768, 199),
 ('Saint Kitts and Nevis', 51936, 261, 198),
 ('Gambia, The', 1967709, 10120, 194),
 ('Malawi', 17964697, 94080, 190),
 ('Qatar', 2194817, 11586, 189),
 ('Uganda', 37101745, 197100, 188),
 ('Andorra', 85580, 468, 182),
 ('Anguilla', 16418, 91, 180),
 ('Kosovo', 1870981, 10887, 171),
 ('Kuwait', 2788534, 17818, 156),
 ('Isle of Man', 87545, 572, 153),
 ('Micronesia, Federated States of', 105216, 702, 149),
 ('Tonga', 106501, 717, 148),
 ('China', 1367485388, 9326410, 146),
 ('Indonesia', 255993674, 1811569, 141),
 ('Guatemala', 14918999, 107159, 139),
 ('Togo', 7552318, 54385, 138),
 ('Czech Republic', 10644842, 77247, 137),
 ('Cabo Verde', 545993, 4033, 135),
 ('Thailand', 67976405, 510890, 133),
 ('Denmark', 5581503, 42434, 131),
 ('Kiribati', 105711, 811, 130),
 ('Cyprus', 1189197, 9241, 128),
 ('Poland', 38562189, 304255, 126),
 ('Azerbaijan', 9780780, 82629, 118),
 ('Portugal', 10825309, 91470, 118),
 ('Ghana', 26327649, 227533, 115),
 ('Slovakia', 5445027, 48105, 113),
 ('Northern Mariana Islands', 52344, 464, 112),
 ('Tokelau', 1337, 12, 111),
 ('Albania', 3029278, 27398, 110),
 ('Hungary', 9897541, 89608, 110),
 ('Wallis and Futuna', 15613, 142, 109),
 ('Armenia', 3056382, 28203, 108),
 ('Moldova', 3546847, 32891, 107),
 ('Austria', 8665550, 82445, 105),
 ('France', 66553766, 640427, 103),
 ('Turkey', 79414269, 769632, 103),
 ('Cuba', 11031433, 109820, 100),
 ('Dominica', 73607, 751, 98),
 ('Slovenia', 1983412, 20151, 98),
 ('Spain', 48146134, 498980, 96),
 ('Benin', 10448647, 110622, 94),
 ('Costa Rica', 4814144, 51060, 94),
 ('Romania', 21666350, 229891, 94),
 ('Malaysia', 30513848, 328

Figure out that "Macau" is the country that have the most density population per land area and "Monaco", "Singapore" and "Hong Kong" respectively.

## Exploring which country have the highest growth rate

In [65]:
%%sql
SELECT name, population_growth
  FROM facts
 WHERE name <> 'World'
 ORDER BY population_growth DESC;

Done.


[('South Sudan', 4.02),
 ('Malawi', 3.32),
 ('Burundi', 3.28),
 ('Niger', 3.25),
 ('Uganda', 3.24),
 ('Qatar', 3.07),
 ('Burkina Faso', 3.03),
 ('Mali', 2.98),
 ('Cook Islands', 2.95),
 ('Iraq', 2.93),
 ('Ethiopia', 2.89),
 ('Zambia', 2.88),
 ('Western Sahara', 2.82),
 ('Gaza Strip', 2.81),
 ('Tanzania', 2.79),
 ('Angola', 2.78),
 ('Benin', 2.78),
 ('Togo', 2.69),
 ('Guinea', 2.63),
 ('Cameroon', 2.59),
 ('Madagascar', 2.58),
 ('United Arab Emirates', 2.58),
 ('Rwanda', 2.56),
 ('Equatorial Guinea', 2.51),
 ('Liberia', 2.47),
 ('Yemen', 2.47),
 ('Congo, Democratic Republic of the', 2.45),
 ('Mozambique', 2.45),
 ('Nigeria', 2.45),
 ('Senegal', 2.45),
 ('Timor-Leste', 2.42),
 ('Bahrain', 2.41),
 ('Sierra Leone', 2.35),
 ('Afghanistan', 2.32),
 ('British Virgin Islands', 2.32),
 ('Turks and Caicos Islands', 2.3),
 ('Eritrea', 2.25),
 ('Libya', 2.23),
 ('Mauritania', 2.23),
 ('Zimbabwe', 2.21),
 ('Djibouti', 2.2),
 ('Ghana', 2.18),
 ('Northern Mariana Islands', 2.18),
 ('Gambia, The', 2.16),
 ('Central African Republic', 2.13),
 ('Luxembourg', 2.13),
 ('Cayman Islands', 2.1),
 ('Oman', 2.07),
 ('Anguilla', 2.03),
 ('Solomon Islands', 2.02),
 ('Congo, Republic of the', 2.0),
 ('Vanuatu', 1.95),
 ('West Bank', 1.95),
 ('Gabon', 1.93),
 ('Kenya', 1.93),
 ("Cote d'Ivoire", 1.91),
 ('Guinea-Bissau', 1.91),
 ('Chad', 1.89),
 ('Singapore', 1.89),
 ('Belize', 1.87),
 ('Algeria', 1.84),
 ('Sao Tome and Principe', 1.84),
 ('Somalia', 1.83),
 ('Guatemala', 1.82),
 ('Egypt', 1.79),
 ('Nepal', 1.79),
 ('Papua New Guinea', 1.78),
 ('Comoros', 1.77),
 ('Sudan', 1.72),
 ('Tajikistan', 1.71),
 ('Honduras', 1.68),
 ('Marshall Islands', 1.66),
 ('Brunei', 1.62),
 ('Kuwait', 1.62),
 ('Philippines', 1.61),
 ('Bangladesh', 1.6),
 ('Cambodia', 1.58),
 ('Bolivia', 1.56),
 ('Israel', 1.56),
 ('Laos', 1.55),
 ('Sint Maarten', 1.51),
 ('Pakistan', 1.46),
 ('Saudi Arabia', 1.46),
 ('Malaysia', 1.44),
 ('Cyprus', 1.43),
 ('Venezuela', 1.39),
 ('New Caledonia', 1.38),
 ('Cabo Verde', 1.36),
 ('Ecuador', 1.35),
 ('South Africa', 1.33),
 ('Aruba', 1.33),
 ('Panama', 1.32),
 ('Mongolia', 1.31),
 ('Turkey', 1.26),
 ('Ireland', 1.25),
 ('Antigua and Barbuda', 1.24),
 ('Dominican Republic', 1.23),
 ('Costa Rica', 1.22),
 ('India', 1.22),
 ('Botswana', 1.21),
 ('Iceland', 1.21),
 ('Iran', 1.2),
 ('Mexico', 1.18),
 ('Haiti', 1.17),
 ('Paraguay', 1.16),
 ('Kiribati', 1.15),
 ('Kazakhstan', 1.14),
 ('Turkmenistan', 1.14),
 ('Norway', 1.13),
 ('Bhutan', 1.11),
 ('Kyrgyzstan', 1.11),
 ('Swaziland', 1.11),
 ('Christmas Island', 1.11),
 ('Suriname', 1.08),
 ('Saint Pierre and Miquelon', 1.08),
 ('Australia', 1.07),
 ('Latvia', 1.06),
 ('Colombia', 1.04),
 ('Lithuania', 1.04),
 ('Moldova', 1.03),
 ('Burma', 1.01),
 ('Morocco', 1.0),
 ('Nicaragua', 1.0),
 ('Peru', 0.97),
 ('Vietnam', 0.97),
 ('Azerbaijan', 0.96),
 ('French Polynesia', 0.94),
 ('Argentina', 0.93),
 ('Uzbekistan', 0.93),
 ('Indonesia', 0.92),
 ('Spain', 0.89),
 ('Tunisia', 0.89),
 ('Lebanon', 0.86),
 ('Bahamas, The', 0.85),
 ('Liechtenstein', 0.84),
 ('Sri Lanka', 0.84),
 ('Jordan', 0.83),
 ('Seychelles', 0.83),
 ('Chile', 0.82),
 ('New Zealand', 0.82),
 ('San Marino', 0.82),
 ('Tuvalu', 0.82),
 ('Sweden', 0.8),
 ('Macau', 0.8),
 ('Jersey', 0.8),
 ('United States', 0.78),
 ('Brazil', 0.77),
 ('Belgium', 0.76),
 ('Saint Kitts and Nevis', 0.76),
 ('Isle of Man', 0.76),
 ('Canada', 0.75),
 ('Switzerland', 0.71),
 ('Jamaica', 0.68),
 ('Fiji', 0.67),
 ('Mauritius', 0.64),
 ('Ukraine', 0.6),
 ('Puerto Rico', 0.6),
 ('Namibia', 0.59),
 ('Virgin Islands', 0.59),
 ('Bulgaria', 0.58),
 ('Samoa', 0.58),
 ('Austria', 0.55),
 ('Estonia', 0.55),
 ('Nauru', 0.55),
 ('United Kingdom', 0.54),
 ('Guam', 0.54),
 ('Korea, North', 0.53),
 ('Faroe Islands', 0.51),
 ('Bermuda', 0.5),
 ('Montserrat', 0.5),
 ('Grenada', 0.48),
 ('Micronesia, Federated States of', 0.46),
 ('Serbia', 0.46),
 ('China', 0.45),
 ('France', 0.43),
 ('Curacao', 0.43),
 ('Montenegro', 0.42),
 ('Netherlands', 0.41),
 ('Finland', 0.4),
 ('Palau', 0.38),
 ('Hong

Figure out that "South Sudan" is the country that have to most population growth rate that found in the earlier queries.

Let's find which country will add the most people to thier populations next year?

In [67]:
%%sql
SELECT name, population, population_growth,
       population * population_growth AS next_year_pop
  FROM facts
 WHERE name <> 'World'
 ORDER BY next_year_pop DESC;

Done.


[('India', 1251695584, 1.22, 1527068612.48),
 ('China', 1367485388, 0.45, 615368424.6),
 ('Nigeria', 181562056, 2.45, 444827037.20000005),
 ('Pakistan', 199085847, 1.46, 290665336.62),
 ('Ethiopia', 99465819, 2.89, 287456216.91),
 ('Bangladesh', 168957745, 1.6, 270332392.0),
 ('United States', 321368864, 0.78, 250667713.92000002),
 ('Indonesia', 255993674, 0.92, 235514180.08),
 ('Congo, Democratic Republic of the', 79375136, 2.45, 194469083.20000002),
 ('Philippines', 100998376, 1.61, 162607385.36),
 ('Egypt', 88487396, 1.79, 158392438.84),
 ('Brazil', 204259812, 0.77, 157280055.24),
 ('Mexico', 121736809, 1.18, 143649434.62),
 ('Tanzania', 51045882, 2.79, 142418010.78),
 ('European Union', 513949445, 0.25, 128487361.25),
 ('Uganda', 37101745, 3.24, 120209653.80000001),
 ('Iraq', 37056169, 2.93, 108574575.17),
 ('Turkey', 79414269, 1.26, 100061978.94),
 ('Iran', 81824270, 1.2, 98189124.0),
 ('Vietnam', 94348835, 0.97, 91518369.95),
 ('Kenya', 45925301, 1.93, 88635830.92999999),
 ('Afghanistan', 32564342, 2.32, 75549273.44),
 ('Algeria', 39542166, 1.84, 72757585.44),
 ('South Africa', 53675563, 1.33, 71388498.79),
 ('Yemen', 26737317, 2.47, 66041172.99),
 ('Sudan', 36108853, 1.72, 62107227.16),
 ('Mozambique', 25303113, 2.45, 61992626.85),
 ('Cameroon', 23739218, 2.59, 61484574.62),
 ('Madagascar', 23812681, 2.58, 61436716.980000004),
 ('Malawi', 17964697, 3.32, 59642794.04),
 ('Niger', 18045729, 3.25, 58648619.25),
 ('Ghana', 26327649, 2.18, 57394274.82000001),
 ('Burkina Faso', 18931686, 3.03, 57363008.58),
 ('Burma', 56320206, 1.01, 56883408.06),
 ('Nepal', 31551305, 1.79, 56476835.95),
 ('Angola', 19625353, 2.78, 54558481.339999996),
 ('Mali', 16955536, 2.98, 50527497.28),
 ('Colombia', 46736728, 1.04, 48606197.120000005),
 ('South Sudan', 12042910, 4.02, 48412498.199999996),
 ("Cote d'Ivoire", 23295302, 1.91, 44494026.82),
 ('Malaysia', 30513848, 1.44, 43939941.12),
 ('Zambia', 15066266, 2.88, 43390846.08),
 ('Spain', 48146134, 0.89, 42850059.26),
 ('Venezuela', 29275460, 1.39, 40692889.4),
 ('Saudi Arabia', 27752316, 1.46, 40518381.36),
 ('Argentina', 43431886, 0.93, 40391653.980000004),
 ('Burundi', 10742276, 3.28, 35234665.28),
 ('United Kingdom', 64088222, 0.54, 34607639.88),
 ('Senegal', 13975834, 2.45, 34240793.300000004),
 ('Morocco', 33322699, 1.0, 33322699.0),
 ('Rwanda', 12661733, 2.56, 32414036.48),
 ('Zimbabwe', 14229541, 2.21, 31447285.61),
 ('Guinea', 11780162, 2.63, 30981826.06),
 ('Peru', 30444999, 0.97, 29531649.029999997),
 ('Benin', 10448647, 2.78, 29047238.659999996),
 ('France', 66553766, 0.43, 28618119.38),
 ('Uzbekistan', 29199942, 0.93, 27155946.060000002),
 ('Guatemala', 14918999, 1.82, 27152578.18),
 ('Ukraine', 44429471, 0.6, 26657682.599999998),
 ('Canada', 35099836, 0.75, 26324877.0),
 ('Cambodia', 15708756, 1.58, 24819834.48),
 ('Australia', 22751014, 1.07, 24343584.98),
 ('Thailand', 67976405, 0.34, 23111977.700000003),
 ('Chad', 11631456, 1.89, 21983451.84),
 ('Ecuador', 15868396, 1.35, 21422334.6),
 ('Kazakhstan', 18157122, 1.14, 20699119.08),
 ('Togo', 7552318, 2.69, 20315735.419999998),
 ('Japan', 126919659, 0.16, 20307145.44),
 ('Somalia', 10616380, 1.83, 19427975.400000002),
 ('Sri Lanka', 22053488, 0.84, 18524929.919999998),
 ('Bolivia', 10800882, 1.56, 16849375.92),
 ('Italy', 61855120, 0.27, 16700882.4),
 ('United Arab Emirates', 5779760, 2.58, 14911780.8),
 ('Honduras', 8746673, 1.68, 14694410.639999999),
 ('Eritrea', 6527689, 2.25, 14687300.25),
 ('Chile', 17508260, 0.82, 14356773.2),
 ('Libya', 6411776, 2.23, 14298260.48),
 ('Tajikistan', 8191958, 1.71, 14008248.18),
 ('Sierra Leone', 5879098, 2.35, 13815880.3),
 ('Germany', 80854408, 0.17, 13745249.360000001),
 ('Korea, North', 24983205, 0.53, 13241098.65),
 ('Dominican Republic', 10478756, 1.23, 12888869.879999999),
 ('Israel', 8049314, 1.56, 12556929.84),
 ('Papua New Guinea', 6672429, 1.78, 11876923.620000001),
 ('Haiti', 10110019, 1.17, 11828722.229999999),
 ('Central African Republic', 5391539, 2.13, 11483978.07),
 ('Singapo

If multiply the current year of population with population growth in the same year "India" will be the country in the next year that is the most population in the world.

## Exploring which country have a higher death rate than birth rate

In [88]:
%%sql
SELECT name, birth_rate, death_rate,
       ROUND(death_rate - birth_rate, 2) AS diff
  FROM facts
 WHERE name <> 'World'
   AND birth_rate < death_rate
 ORDER BY diff DESC;

Done.


[('Bulgaria', 8.92, 14.44, 5.52),
 ('Serbia', 9.08, 13.66, 4.58),
 ('Latvia', 10.0, 14.31, 4.31),
 ('Lithuania', 10.1, 14.27, 4.17),
 ('Ukraine', 10.72, 14.46, 3.74),
 ('Hungary', 9.16, 12.73, 3.57),
 ('Germany', 8.47, 11.42, 2.95),
 ('Slovenia', 8.42, 11.37, 2.95),
 ('Romania', 9.14, 11.9, 2.76),
 ('Croatia', 9.45, 12.18, 2.73),
 ('Belarus', 10.7, 13.36, 2.66),
 ('Monaco', 6.65, 9.24, 2.59),
 ('Greece', 8.66, 11.09, 2.43),
 ('Saint Pierre and Miquelon', 7.42, 9.72, 2.3),
 ('Russia', 11.6, 13.69, 2.09),
 ('Estonia', 10.51, 12.4, 1.89),
 ('Portugal', 9.27, 11.02, 1.75),
 ('Japan', 7.93, 9.51, 1.58),
 ('Italy', 8.74, 10.19, 1.45),
 ('Bosnia and Herzegovina', 8.87, 9.75, 0.88),
 ('Czech Republic', 9.63, 10.34, 0.71),
 ('Moldova', 12.0, 12.59, 0.59),
 ('Poland', 9.74, 10.19, 0.45),
 ('Austria', 9.41, 9.42, 0.01)]

Another topic that is which countries that have a higher death rate than birth rate!!

For the answer is "Bulgaria"

## Exploring about water area and land area
1) which countries have the highest ratios of water to land? 

2) which countries have more water than land?

Let's explore the data..

In [90]:
%%sql
SELECT name, area_water
  FROM facts
 WHERE name <> 'World'
 ORDER BY area_water DESC;

Done.


[('Canada', 891163),
 ('Russia', 720500),
 ('United States', 664709),
 ('India', 314070),
 ('China', 270550),
 ('Brazil', 157630),
 ('Iran', 116600),
 ('Ethiopia', 104300),
 ('Colombia', 100210),
 ('Indonesia', 93000),
 ('Congo, Democratic Republic of the', 77810),
 ('Tanzania', 61500),
 ('Australia', 58920),
 ('British Indian Ocean Territory', 54340),
 ('Uganda', 43938),
 ('Argentina', 43710),
 ('Sweden', 39960),
 ('Finland', 34330),
 ('Venezuela', 30000),
 ('Pakistan', 25220),
 ('Kazakhstan', 25200),
 ('Chad', 24800),
 ('Malawi', 24404),
 ('Ukraine', 24220),
 ('Burma', 23070),
 ('Uzbekistan', 22000),
 ('Vietnam', 21140),
 ('Mexico', 20430),
 ('Mali', 20002),
 ('Norway', 19520),
 ('Bangladesh', 18290),
 ('Turkmenistan', 18170),
 ('Guyana', 18120),
 ('Eritrea', 16600),
 ('Bolivia', 15280),
 ('Liberia', 15049),
 ('Botswana', 15000),
 ('Turkey', 13930),
 ('Japan', 13430),
 ('Mozambique', 13000),
 ('Nigeria', 13000),
 ('Chile', 12290),
 ('Kenya', 11227),
 ('Ghana', 11000),
 ('Mongolia', 10560),
 ('Nicaragua', 10380),
 ('Somalia', 10320),
 ('Gabon', 10000),
 ('Papua New Guinea', 9980),
 ('Paraguay', 9450),
 ('Zambia', 9220),
 ('Romania', 8500),
 ('Poland', 8430),
 ('Germany', 8350),
 ('Tunisia', 8250),
 ('Kyrgyzstan', 8150),
 ('Guinea-Bissau', 8005),
 ('Suriname', 7820),
 ('Netherlands', 7650),
 ('Italy', 7200),
 ('Ecuador', 6720),
 ('Spain', 6390),
 ('Egypt', 6000),
 ('Laos', 6000),
 ('Madagascar', 5501),
 ('Peru', 5220),
 ('Puerto Rico', 4921),
 ('Belarus', 4700),
 ('South Africa', 4620),
 ('Cambodia', 4520),
 ("Cote d'Ivoire", 4460),
 ('Senegal', 4192),
 ('Azerbaijan', 3971),
 ('Zimbabwe', 3910),
 ('Bahamas, The', 3870),
 ('Nepal', 3830),
 ('Taiwan', 3720),
 ('Hungary', 3420),
 ('France', 3374),
 ('Estonia', 2840),
 ('Korea, South', 2800),
 ('Iceland', 2750),
 ('Cameroon', 2730),
 ('Lithuania', 2620),
 ('Tajikistan', 2590),
 ('Togo', 2400),
 ('Bulgaria', 2390),
 ('Latvia', 2340),
 ('Thailand', 2230),
 ('Burundi', 2150),
 ('Benin', 2000),
 ('Philippines', 1830),
 ('Guatemala', 1730),
 ('United Kingdom', 1680),
 ('Rwanda', 1670),
 ('Czech Republic', 1620),
 ('Virgin Islands', 1564),
 ('Syria', 1550),
 ('Armenia', 1540),
 ('Austria', 1426),
 ('Ireland', 1390),
 ('Albania', 1350),
 ('Greece', 1310),
 ('Switzerland', 1280),
 ('Uruguay', 1200),
 ('Malaysia', 1190),
 ('Gambia, The', 1180),
 ('Panama', 1080),
 ('Cuba', 1040),
 ('Namibia', 1002),
 ('Sri Lanka', 980),
 ('Moldova', 960),
 ('Iraq', 950),
 ('Slovakia', 930),
 ('Solomon Islands', 910),
 ('Denmark', 660),
 ('Croatia', 620),
 ('Portugal', 620),
 ('Jordan', 540),
 ('Brunei', 500),
 ('Congo, Republic of the', 500),
 ('Israel', 440),
 ('Burkina Faso', 400),
 ('Montenegro', 360),
 ('Dominican Republic', 350),
 ('French Polynesia', 340),
 ('El Salvador', 320),
 ('Niger', 300),
 ('New Caledonia', 300),
 ('Macedonia', 280),
 ('Belgium', 250),
 ('Morocco', 250),
 ('West Bank', 220),
 ('Honduras', 200),
 ('Haiti', 190),
 ('Lebanon', 170),
 ('Belize', 160),
 ('Jamaica', 160),
 ('Swaziland', 160),
 ('Guinea', 140),
 ('Korea, North', 130),
 ('Slovenia', 122),
 ('Sierra Leone', 120),
 ('Costa Rica', 40),
 ('Hong Kong', 35),
 ('Tonga', 30),
 ('Djibouti', 20),
 ('Bosnia and Herzegovina', 10),
 ('Cyprus', 10),
 ('Mauritius', 10),
 ('Saint Lucia', 10),
 ('Samoa', 10),
 ('Singapore', 10),
 ('Afghanistan', 0),
 ('Algeria', 0),
 ('Andorra', 0),
 ('Angola', 0),
 ('Antigua and Barbuda', 0),
 ('Bahrain', 0),
 ('Barbados', 0),
 ('Bhutan', 0),
 ('Cabo Verde', 0),
 ('Central African Republic', 0),
 ('Comoros', 0),
 ('Dominica', 0),
 ('Equatorial Guinea', 0),
 ('Fiji', 0),
 ('Georgia', 0),
 ('Grenada', 0),
 ('Kiribati', 0),
 ('Kosovo', 0),
 ('Kuwait', 0),
 ('Lesotho', 0),
 ('Libya', 0),
 ('Liechtenstein', 0),
 ('Luxembourg', 0),
 ('Maldives', 0),
 ('Malta', 0),
 ('Marshall Islands', 0),
 ('Mauritania', 0),
 ('Micronesia, Federated States of', 0),
 ('Monaco', 0),
 ('Nauru', 0),
 ('Oman', 0),
 ('Palau', 0),
 ('Qatar', 0),
 ('Saint Kitts and Nevis', 0),
 ('Saint Vincent and the Grenadines', 0),
 ('San Mari

In [91]:
%%sql
SELECT MIN(area_water),
       MAX(area_water)
  FROM facts
 WHERE name <> 'World'

Done.


[(0, 891163)]

There's some countries with area water with 0.

Let's explore more to find insight.

In [93]:
%%sql
SELECT name, area_water
  FROM facts
 WHERE area_water = (SELECT MIN(area_water)
                       FROM facts
                      WHERE name <> 'World');

Done.


[('Afghanistan', 0),
 ('Algeria', 0),
 ('Andorra', 0),
 ('Angola', 0),
 ('Antigua and Barbuda', 0),
 ('Bahrain', 0),
 ('Barbados', 0),
 ('Bhutan', 0),
 ('Cabo Verde', 0),
 ('Central African Republic', 0),
 ('Comoros', 0),
 ('Dominica', 0),
 ('Equatorial Guinea', 0),
 ('Fiji', 0),
 ('Georgia', 0),
 ('Grenada', 0),
 ('Kiribati', 0),
 ('Kosovo', 0),
 ('Kuwait', 0),
 ('Lesotho', 0),
 ('Libya', 0),
 ('Liechtenstein', 0),
 ('Luxembourg', 0),
 ('Maldives', 0),
 ('Malta', 0),
 ('Marshall Islands', 0),
 ('Mauritania', 0),
 ('Micronesia, Federated States of', 0),
 ('Monaco', 0),
 ('Nauru', 0),
 ('Oman', 0),
 ('Palau', 0),
 ('Qatar', 0),
 ('Saint Kitts and Nevis', 0),
 ('Saint Vincent and the Grenadines', 0),
 ('San Marino', 0),
 ('Sao Tome and Principe', 0),
 ('Saudi Arabia', 0),
 ('Serbia', 0),
 ('Seychelles', 0),
 ('Timor-Leste', 0),
 ('Trinidad and Tobago', 0),
 ('Tuvalu', 0),
 ('United Arab Emirates', 0),
 ('Vanuatu', 0),
 ('Holy See (Vatican City)', 0),
 ('Yemen', 0),
 ('Ashmore and Cartier Islands', 0),
 ('Christmas Island', 0),
 ('Cocos (Keeling) Islands', 0),
 ('Coral Sea Islands', 0),
 ('Heard Island and McDonald Islands', 0),
 ('Norfolk Island', 0),
 ('Macau', 0),
 ('Faroe Islands', 0),
 ('Clipperton Island', 0),
 ('Saint Pierre and Miquelon', 0),
 ('Wallis and Futuna', 0),
 ('Aruba', 0),
 ('Curacao', 0),
 ('Sint Maarten', 0),
 ('Cook Islands', 0),
 ('Niue', 0),
 ('Tokelau', 0),
 ('Bouvet Island', 0),
 ('Jan Mayen', 0),
 ('Svalbard', 0),
 ('Anguilla', 0),
 ('Bermuda', 0),
 ('British Virgin Islands', 0),
 ('Cayman Islands', 0),
 ('Falkland Islands (Islas Malvinas)', 0),
 ('Gibraltar', 0),
 ('Guernsey', 0),
 ('Jersey', 0),
 ('Isle of Man', 0),
 ('Montserrat', 0),
 ('Pitcairn Islands', 0),
 ('Saint Helena, Ascension, and Tristan da Cunha', 0),
 ('South Georgia and South Sandwich Islands', 0),
 ('Turks and Caicos Islands', 0),
 ('American Samoa', 0),
 ('Guam', 0),
 ('Navassa Island', 0),
 ('Northern Mariana Islands', 0),
 ('Wake Island', 0),
 ('Gaza Strip', 0),
 ('Paracel Islands', 0),
 ('Spratly Islands', 0),
 ('Western Sahara', 0)]

In [97]:
%%sql
SELECT COUNT(*) AS no_water_area
  FROM facts
 WHERE area_water = (SELECT MIN(area_water)
                       FROM facts
                      WHERE name <> 'World');

Done.


[(90,)]

There are 90 countries that don't have area water in thier countries.

In [98]:
%%sql
SELECT name, area_land
  FROM facts
 WHERE name <> 'World'
 ORDER BY area_land DESC;

Done.


[('Russia', 16377742),
 ('China', 9326410),
 ('United States', 9161966),
 ('Canada', 9093507),
 ('Brazil', 8358140),
 ('Australia', 7682300),
 ('India', 2973193),
 ('Argentina', 2736690),
 ('Kazakhstan', 2699700),
 ('Algeria', 2381741),
 ('Congo, Democratic Republic of the', 2267048),
 ('Greenland', 2166086),
 ('Saudi Arabia', 2149690),
 ('Mexico', 1943945),
 ('Indonesia', 1811569),
 ('Libya', 1759540),
 ('Mongolia', 1553556),
 ('Iran', 1531595),
 ('Peru', 1279996),
 ('Niger', 1266700),
 ('Chad', 1259200),
 ('Angola', 1246700),
 ('Mali', 1220190),
 ('South Africa', 1214470),
 ('Bolivia', 1083301),
 ('Colombia', 1038700),
 ('Mauritania', 1030700),
 ('Egypt', 995450),
 ('Nigeria', 910768),
 ('Tanzania', 885800),
 ('Venezuela', 882050),
 ('Namibia', 823290),
 ('Mozambique', 786380),
 ('Pakistan', 770875),
 ('Turkey', 769632),
 ('Chile', 743812),
 ('Zambia', 743398),
 ('Burma', 653508),
 ('Afghanistan', 652230),
 ('France', 640427),
 ('Somalia', 627337),
 ('Central African Republic', 622984),
 ('Madagascar', 581540),
 ('Ukraine', 579330),
 ('Kenya', 569140),
 ('Botswana', 566730),
 ('Yemen', 527968),
 ('Thailand', 510890),
 ('Spain', 498980),
 ('Cameroon', 472710),
 ('Turkmenistan', 469930),
 ('Papua New Guinea', 452860),
 ('Morocco', 446300),
 ('Iraq', 437367),
 ('Uzbekistan', 425400),
 ('Sweden', 410335),
 ('Paraguay', 397302),
 ('Zimbabwe', 386847),
 ('Japan', 364485),
 ('Germany', 348672),
 ('Congo, Republic of the', 341500),
 ('Malaysia', 328657),
 ("Cote d'Ivoire", 318003),
 ('Vietnam', 310070),
 ('Oman', 309500),
 ('Norway', 304282),
 ('Poland', 304255),
 ('Finland', 303815),
 ('Philippines', 298170),
 ('Italy', 294140),
 ('Antarctica', 280000),
 ('Ecuador', 276841),
 ('Burkina Faso', 273800),
 ('New Zealand', 267710),
 ('Western Sahara', 266000),
 ('Gabon', 257667),
 ('Guinea', 245717),
 ('United Kingdom', 241930),
 ('Laos', 230800),
 ('Romania', 229891),
 ('Ghana', 227533),
 ('Belarus', 202900),
 ('Uganda', 197100),
 ('Guyana', 196849),
 ('Senegal', 192530),
 ('Kyrgyzstan', 191801),
 ('Syria', 183630),
 ('Cambodia', 176515),
 ('Uruguay', 175015),
 ('Suriname', 156000),
 ('Tunisia', 155360),
 ('Nepal', 143351),
 ('Tajikistan', 141510),
 ('Greece', 130647),
 ('Bangladesh', 130170),
 ('Korea, North', 120408),
 ('Nicaragua', 119990),
 ('Honduras', 111890),
 ('Benin', 110622),
 ('Cuba', 109820),
 ('Bulgaria', 108489),
 ('Guatemala', 107159),
 ('Eritrea', 101000),
 ('Iceland', 100250),
 ('Korea, South', 96920),
 ('Liberia', 96320),
 ('Malawi', 94080),
 ('Portugal', 91470),
 ('Hungary', 89608),
 ('Jordan', 88802),
 ('United Arab Emirates', 83600),
 ('Azerbaijan', 82629),
 ('Austria', 82445),
 ('Serbia', 77474),
 ('Czech Republic', 77247),
 ('Panama', 74340),
 ('Sierra Leone', 71620),
 ('Georgia', 69700),
 ('Ireland', 68883),
 ('Sri Lanka', 64630),
 ('Lithuania', 62680),
 ('Latvia', 62249),
 ('Svalbard', 62045),
 ('Croatia', 55974),
 ('Togo', 54385),
 ('Bosnia and Herzegovina', 51187),
 ('Costa Rica', 51060),
 ('Dominican Republic', 48320),
 ('Slovakia', 48105),
 ('Denmark', 42434),
 ('Estonia', 42388),
 ('Switzerland', 39997),
 ('Bhutan', 38394),
 ('Netherlands', 33893),
 ('Moldova', 32891),
 ('Taiwan', 32260),
 ('Lesotho', 30355),
 ('Belgium', 30278),
 ('Armenia', 28203),
 ('Guinea-Bissau', 28120),
 ('Equatorial Guinea', 28051),
 ('Solomon Islands', 27986),
 ('Haiti', 27560),
 ('Albania', 27398),
 ('Burundi', 25680),
 ('Macedonia', 25433),
 ('Rwanda', 24668),
 ('Djibouti', 23180),
 ('Belize', 22806),
 ('El Salvador', 20721),
 ('Israel', 20330),
 ('Slovenia', 20151),
 ('New Caledonia', 18275),
 ('Fiji', 18274),
 ('Kuwait', 17818),
 ('Swaziland', 17204),
 ('Timor-Leste', 14874),
 ('Montenegro', 13452),
 ('Vanuatu', 12189),
 ('Falkland Islands (Islas Malvinas)', 12173),
 ('Qatar', 11586),
 ('Kosovo', 10887),
 ('Jamaica', 10831),
 ('Lebanon', 10230),
 ('Gambia, The', 10120),
 ('Bahamas, The', 10010),
 ('Cyprus', 9241),
 ('Puerto Rico', 8870),
 ('West Bank', 5640),
 ('Brunei', 5265),
 ('Trinidad and Tobago', 5128),
 ('Cabo Verde', 4

### Conclude

 - For the country that have the most area of land is "Russia".
 - For the country that have the most area of water is "Canada".
 
Let's find out which countries that have area of water than land

In [99]:
%%sql
SELECT name,
       area_water - area_land AS diff_area
  FROM facts
 WHERE name <> 'World'
 ORDER BY diff_area DESC;

Done.


[('British Indian Ocean Territory', 54280),
 ('Virgin Islands', 1218),
 ('Holy See (Vatican City)', 0),
 ('Monaco', -2),
 ('Coral Sea Islands', -3),
 ('Ashmore and Cartier Islands', -5),
 ('Navassa Island', -5),
 ('Spratly Islands', -5),
 ('Clipperton Island', -6),
 ('Gibraltar', -6),
 ('Wake Island', -6),
 ('Paracel Islands', -7),
 ('Tokelau', -12),
 ('Cocos (Keeling) Islands', -14),
 ('Nauru', -21),
 ('Tuvalu', -26),
 ('Macau', -28),
 ('Sint Maarten', -34),
 ('Norfolk Island', -36),
 ('Pitcairn Islands', -47),
 ('Bouvet Island', -49),
 ('Bermuda', -54),
 ('San Marino', -61),
 ('Guernsey', -78),
 ('Anguilla', -91),
 ('Montserrat', -102),
 ('Jersey', -116),
 ('Saint Helena, Ascension, and Tristan da Cunha', -122),
 ('Christmas Island', -135),
 ('Wallis and Futuna', -142),
 ('British Virgin Islands', -151),
 ('Liechtenstein', -160),
 ('Aruba', -180),
 ('Marshall Islands', -181),
 ('American Samoa', -199),
 ('Cook Islands', -236),
 ('Saint Pierre and Miquelon', -242),
 ('Niue', -260),
 ('Saint Kitts and Nevis', -261),
 ('Cayman Islands', -264),
 ('Maldives', -298),
 ('Malta', -316),
 ('Grenada', -344),
 ('Gaza Strip', -360),
 ('Jan Mayen', -377),
 ('Saint Vincent and the Grenadines', -389),
 ('Heard Island and McDonald Islands', -412),
 ('Barbados', -430),
 ('Antigua and Barbuda', -442),
 ('Curacao', -444),
 ('Seychelles', -455),
 ('Palau', -459),
 ('Northern Mariana Islands', -464),
 ('Andorra', -468),
 ('Guam', -544),
 ('Isle of Man', -572),
 ('Saint Lucia', -596),
 ('Singapore', -677),
 ('Tonga', -687),
 ('Micronesia, Federated States of', -702),
 ('Dominica', -751),
 ('Bahrain', -760),
 ('Kiribati', -811),
 ('Turks and Caicos Islands', -948),
 ('Sao Tome and Principe', -964),
 ('Hong Kong', -1038),
 ('Faroe Islands', -1393),
 ('Mauritius', -2020),
 ('Comoros', -2235),
 ('Luxembourg', -2586),
 ('Samoa', -2811),
 ('French Polynesia', -3487),
 ('South Georgia and South Sandwich Islands', -3903),
 ('Puerto Rico', -3949),
 ('Cabo Verde', -4033),
 ('Brunei', -4765),
 ('Trinidad and Tobago', -5128),
 ('West Bank', -5420),
 ('Bahamas, The', -6140),
 ('Gambia, The', -8940),
 ('Cyprus', -9231),
 ('Lebanon', -10060),
 ('Jamaica', -10671),
 ('Kosovo', -10887),
 ('Qatar', -11586),
 ('Falkland Islands (Islas Malvinas)', -12173),
 ('Vanuatu', -12189),
 ('Montenegro', -13092),
 ('Timor-Leste', -14874),
 ('Swaziland', -17044),
 ('Kuwait', -17818),
 ('New Caledonia', -17975),
 ('Fiji', -18274),
 ('Israel', -19890),
 ('Slovenia', -20029),
 ('Guinea-Bissau', -20115),
 ('El Salvador', -20401),
 ('Belize', -22646),
 ('Rwanda', -22998),
 ('Djibouti', -23160),
 ('Burundi', -23530),
 ('Macedonia', -25153),
 ('Albania', -26048),
 ('Netherlands', -26243),
 ('Armenia', -26663),
 ('Solomon Islands', -27076),
 ('Haiti', -27370),
 ('Equatorial Guinea', -28051),
 ('Taiwan', -28540),
 ('Belgium', -30028),
 ('Lesotho', -30355),
 ('Moldova', -31931),
 ('Bhutan', -38394),
 ('Switzerland', -38717),
 ('Estonia', -39548),
 ('Denmark', -41774),
 ('Slovakia', -47175),
 ('Dominican Republic', -47970),
 ('Costa Rica', -51020),
 ('Bosnia and Herzegovina', -51177),
 ('Togo', -51985),
 ('Croatia', -55354),
 ('Latvia', -59909),
 ('Lithuania', -60060),
 ('Svalbard', -62045),
 ('Sri Lanka', -63650),
 ('Ireland', -67493),
 ('Malawi', -69676),
 ('Georgia', -69700),
 ('Sierra Leone', -71500),
 ('Panama', -73260),
 ('Czech Republic', -75627),
 ('Serbia', -77474),
 ('Azerbaijan', -78658),
 ('Austria', -81019),
 ('Liberia', -81271),
 ('United Arab Emirates', -83600),
 ('Eritrea', -84400),
 ('Hungary', -86188),
 ('Jordan', -88262),
 ('Portugal', -90850),
 ('Korea, South', -94120),
 ('Iceland', -97500),
 ('Guatemala', -105429),
 ('Bulgaria', -106099),
 ('Benin', -108622),
 ('Cuba', -108780),
 ('Nicaragua', -109610),
 ('Honduras', -111690),
 ('Bangladesh', -111880),
 ('Korea, North', -120278),
 ('Greece', -129337),
 ('Tajikistan', -138920),
 ('Nepal', -139521),
 ('Tunisia', -147110),
 ('Suriname', -148180),
 ('Uganda', -153162),
 ('Cambodia', -171995),
 ('Uruguay', -173815),
 ('Guy

For the answer is "British Indian Ocean Territory" is the country that have more water than land.